spacetx/starfish
# [TST] migrate to pytest

- migrate to pytest
- turn off F811 which complains about redefinition of parameters -- we import fixtures in pytest, which get used as parameters, which triggers this warning in every testing file
- update all tests except ISS to use pytest framework
- add merfish testing data and merfish tests that mirror the merfish notebook (just exercising code paths, no scientific validation yet)
- create test fixtures for datasets so they're downloaded once, but can be used multiple times
- remove some ill-fated type hinting in the gaussian filter modules that have bad ducktyping and play poorly with mypy. 

Potential problem: not clear if code coverage is being calculated in the same way as last time, but that trigger appears to only happen on PR so we'll see!
